In [ ]:
import pandas as pd

In [ ]:
covid = pd.read_csv('covid.csv', sep=';')
covid

regiao estado municipio  coduf    codmun  codRegiaoSaude  \
0             Brasil    NaN       NaN     76       NaN             NaN   
1             Brasil    NaN       NaN     76       NaN             NaN   
2             Brasil    NaN       NaN     76       NaN             NaN   
3             Brasil    NaN       NaN     76       NaN             NaN   
4             Brasil    NaN       NaN     76       NaN             NaN   
...              ...    ...       ...    ...       ...             ...   
471991  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471992  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471993  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471994  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471995  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   

         nomeRegiaoSaude        data  semanaEpi  populacaoTCU2019  \
0                    NaN  2022-07-01         26       210147125.0   
1                    NaN  2022-07-02         26       210147125.0   
2                    NaN  2022-07-03         27       210147125.0   
3                    NaN  2022-07-04         27       210147125.0   
4                    NaN  2022-07-05         27       210147125.0   
...                  ...         ...        ...               ...   
471991  DISTRITO FEDERAL  2022-09-18         38         3015268.0   
471992  DISTRITO FEDERAL  2022-09-19         38         3015268.0   
471993  DISTRITO FEDERAL  2022-09-20         38         3015268.0   
471994  DISTRITO FEDERAL  2022-09-21         38         3015268.0   
471995  DISTRITO FEDERAL  2022-09-22         38         3015268.0   

        casosAcumulado  casosNovos  obitosAcumulado  obitosNovos  \
0             32434063       76045           671700          284   
1             32471847       37784           671858          158   
2             32490422       18575           671911           53   
3             32535923       45501           672033          122   
4             32610514       74591           672429          396   
...                ...         ...              ...          ...   
471991          838413           0            11825            0   
471992          838501          88            11825            0   
471993          838530          29            11828            3   
471994          838573          43            11829            1   
471995          838619          46            11829            0   

        Recuperadosnovos  emAcompanhamentoNovos  interior/metropolitana  
0             30873682.0               888681.0                     NaN  
1             30880584.0               919405.0                     NaN  
2             30906575.0               911936.0                     NaN  
3             30967114.0               896776.0                     NaN  
4             31039055.0               899030.0                     NaN  
...                  ...                    ...                     ...  
471991               NaN                    NaN                     1.0  
471992               NaN                    NaN                     1.0  
471993               NaN                    NaN                     1.0  
471994               NaN                    NaN                     1.0  
471995               NaN                    NaN                     1.0  

[471996 rows x 17 columns]

In [ ]:
covid.loc[0,'regiao']

'Brasil'

In [ ]:
covid.loc[:, ['regiao', 'coduf', 'estado', 'municipio']]


regiao  coduf estado municipio
0             Brasil     76    NaN       NaN
1             Brasil     76    NaN       NaN
2             Brasil     76    NaN       NaN
3             Brasil     76    NaN       NaN
4             Brasil     76    NaN       NaN
...              ...    ...    ...       ...
471991  Centro-Oeste     53     DF  Brasília
471992  Centro-Oeste     53     DF  Brasília
471993  Centro-Oeste     53     DF  Brasília
471994  Centro-Oeste     53     DF  Brasília
471995  Centro-Oeste     53     DF  Brasília

[471996 rows x 4 columns]

In [ ]:
covid.loc[0:10, ['regiao']]

regiao
0   Brasil
1   Brasil
2   Brasil
3   Brasil
4   Brasil
5   Brasil
6   Brasil
7   Brasil
8   Brasil
9   Brasil
10  Brasil

In [ ]:
covid.loc[0:10, :]

regiao estado municipio  coduf  codmun  codRegiaoSaude nomeRegiaoSaude  \
0   Brasil    NaN       NaN     76     NaN             NaN             NaN   
1   Brasil    NaN       NaN     76     NaN             NaN             NaN   
2   Brasil    NaN       NaN     76     NaN             NaN             NaN   
3   Brasil    NaN       NaN     76     NaN             NaN             NaN   
4   Brasil    NaN       NaN     76     NaN             NaN             NaN   
5   Brasil    NaN       NaN     76     NaN             NaN             NaN   
6   Brasil    NaN       NaN     76     NaN             NaN             NaN   
7   Brasil    NaN       NaN     76     NaN             NaN             NaN   
8   Brasil    NaN       NaN     76     NaN             NaN             NaN   
9   Brasil    NaN       NaN     76     NaN             NaN             NaN   
10  Brasil    NaN       NaN     76     NaN             NaN             NaN   

          data  semanaEpi  populacaoTCU2019  casosAcumulado  casosNovos  \
0   2022-07-01         26       210147125.0        32434063       76045   
1   2022-07-02         26       210147125.0        32471847       37784   
2   2022-07-03         27       210147125.0        32490422       18575   
3   2022-07-04         27       210147125.0        32535923       45501   
4   2022-07-05         27       210147125.0        32610514       74591   
5   2022-07-06         27       210147125.0        32687680       77166   
6   2022-07-07         27       210147125.0        32759730       72050   
7   2022-07-08         27       210147125.0        32830844       71114   
8   2022-07-09         27       210147125.0        32874501       43657   
9   2022-07-10         28       210147125.0        32896464       21963   
10  2022-07-11         28       210147125.0        32940507       44043   

    obitosAcumulado  obitosNovos  Recuperadosnovos  emAcompanhamentoNovos  \
0            671700          284        30873682.0               888681.0   
1            671858          158        30880584.0               919405.0   
2            671911           53        30906575.0               911936.0   
3            672033          122        30967114.0               896776.0   
4            672429          396        31039055.0               899030.0   
5            672790          361        31077538.0               937352.0   
6            673073          283        31094805.0               991852.0   
7            673339          266        31119463.0              1038042.0   
8            673554          215        31142535.0              1058412.0   
9            673610           56        31181066.0              1041788.0   
10           673758          148        31257740.0              1009009.0   

    interior/metropolitana  
0                      NaN  
1                      NaN  
2                      NaN  
3                      NaN  
4                      NaN  
5                      NaN  
6                      NaN  
7                      NaN  
8                      NaN  
9                      NaN  
10                     NaN

In [ ]:
covid.loc[:,:]

regiao estado municipio  coduf    codmun  codRegiaoSaude  \
0             Brasil    NaN       NaN     76       NaN             NaN   
1             Brasil    NaN       NaN     76       NaN             NaN   
2             Brasil    NaN       NaN     76       NaN             NaN   
3             Brasil    NaN       NaN     76       NaN             NaN   
4             Brasil    NaN       NaN     76       NaN             NaN   
...              ...    ...       ...    ...       ...             ...   
471991  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471992  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471993  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471994  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471995  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   

         nomeRegiaoSaude        data  semanaEpi  populacaoTCU2019  \
0                    NaN  2022-07-01         26       210147125.0   
1                    NaN  2022-07-02         26       210147125.0   
2                    NaN  2022-07-03         27       210147125.0   
3                    NaN  2022-07-04         27       210147125.0   
4                    NaN  2022-07-05         27       210147125.0   
...                  ...         ...        ...               ...   
471991  DISTRITO FEDERAL  2022-09-18         38         3015268.0   
471992  DISTRITO FEDERAL  2022-09-19         38         3015268.0   
471993  DISTRITO FEDERAL  2022-09-20         38         3015268.0   
471994  DISTRITO FEDERAL  2022-09-21         38         3015268.0   
471995  DISTRITO FEDERAL  2022-09-22         38         3015268.0   

        casosAcumulado  casosNovos  obitosAcumulado  obitosNovos  \
0             32434063       76045           671700          284   
1             32471847       37784           671858          158   
2             32490422       18575           671911           53   
3             32535923       45501           672033          122   
4             32610514       74591           672429          396   
...                ...         ...              ...          ...   
471991          838413           0            11825            0   
471992          838501          88            11825            0   
471993          838530          29            11828            3   
471994          838573          43            11829            1   
471995          838619          46            11829            0   

        Recuperadosnovos  emAcompanhamentoNovos  interior/metropolitana  
0             30873682.0               888681.0                     NaN  
1             30880584.0               919405.0                     NaN  
2             30906575.0               911936.0                     NaN  
3             30967114.0               896776.0                     NaN  
4             31039055.0               899030.0                     NaN  
...                  ...                    ...                     ...  
471991               NaN                    NaN                     1.0  
471992               NaN                    NaN                     1.0  
471993               NaN                    NaN                     1.0  
471994               NaN                    NaN                     1.0  
471995               NaN                    NaN                     1.0  

[471996 rows x 17 columns]

In [ ]:
covid.loc[0:10, 'regiao':'semanaEpi']

regiao estado municipio  coduf  codmun  codRegiaoSaude nomeRegiaoSaude  \
0   Brasil    NaN       NaN     76     NaN             NaN             NaN   
1   Brasil    NaN       NaN     76     NaN             NaN             NaN   
2   Brasil    NaN       NaN     76     NaN             NaN             NaN   
3   Brasil    NaN       NaN     76     NaN             NaN             NaN   
4   Brasil    NaN       NaN     76     NaN             NaN             NaN   
5   Brasil    NaN       NaN     76     NaN             NaN             NaN   
6   Brasil    NaN       NaN     76     NaN             NaN             NaN   
7   Brasil    NaN       NaN     76     NaN             NaN             NaN   
8   Brasil    NaN       NaN     76     NaN             NaN             NaN   
9   Brasil    NaN       NaN     76     NaN             NaN             NaN   
10  Brasil    NaN       NaN     76     NaN             NaN             NaN   

          data  semanaEpi  
0   2022-07-01         26  
1   2022-07-02         26  
2   2022-07-03         27  
3   2022-07-04         27  
4   2022-07-05         27  
5   2022-07-06         27  
6   2022-07-07         27  
7   2022-07-08         27  
8   2022-07-09         27  
9   2022-07-10         28  
10  2022-07-11         28

In [ ]:
coduf_filtrado = covid[covid['coduf'] == 53]
coduf_filtrado

# covid.loc[(covid['estado'] == 'MS') & (covid['regiao'] == 'Centro-Oeste')]

covid.loc[(covid['estado'] == 'MS') | (covid['regiao'] == 'Centro-Oeste')]

regiao estado municipio  coduf    codmun  codRegiaoSaude  \
2016    Centro-Oeste     MS       NaN     50       NaN             NaN   
2017    Centro-Oeste     MS       NaN     50       NaN             NaN   
2018    Centro-Oeste     MS       NaN     50       NaN             NaN   
2019    Centro-Oeste     MS       NaN     50       NaN             NaN   
2020    Centro-Oeste     MS       NaN     50       NaN             NaN   
...              ...    ...       ...    ...       ...             ...   
471991  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471992  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471993  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471994  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471995  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   

         nomeRegiaoSaude        data  semanaEpi  populacaoTCU2019  \
2016                 NaN  2022-07-01         26         2778986.0   
2017                 NaN  2022-07-02         26         2778986.0   
2018                 NaN  2022-07-03         27         2778986.0   
2019                 NaN  2022-07-04         27         2778986.0   
2020                 NaN  2022-07-05         27         2778986.0   
...                  ...         ...        ...               ...   
471991  DISTRITO FEDERAL  2022-09-18         38         3015268.0   
471992  DISTRITO FEDERAL  2022-09-19         38         3015268.0   
471993  DISTRITO FEDERAL  2022-09-20         38         3015268.0   
471994  DISTRITO FEDERAL  2022-09-21         38         3015268.0   
471995  DISTRITO FEDERAL  2022-09-22         38         3015268.0   

        casosAcumulado  casosNovos  obitosAcumulado  obitosNovos  \
2016            550694         847            10626            0   
2017            551310         616            10626            0   
2018            552247         937            10626            0   
2019            552247           0            10626            0   
2020            553206         959            10645           19   
...                ...         ...              ...          ...   
471991          838413           0            11825            0   
471992          838501          88            11825            0   
471993          838530          29            11828            3   
471994          838573          43            11829            1   
471995          838619          46            11829            0   

        Recuperadosnovos  emAcompanhamentoNovos  interior/metropolitana  \
2016                 NaN                    NaN                     NaN   
2017                 NaN                    NaN                     NaN   
2018                 NaN                    NaN                     NaN   
2019                 NaN                    NaN                     NaN   
2020                 NaN                    NaN                     NaN   
...                  ...                    ...                     ...   
471991               NaN                    NaN                     1.0   
471992               NaN                    NaN                     1.0   
471993               NaN                    NaN                     1.0   
471994               NaN                    NaN                     1.0   
471995               NaN                    NaN                     1.0   

       (0, regiao)  
2016         teste  
2017         teste  
2018         teste  
2019         teste  
2020         teste  
...            ...  
471991       teste  
471992       teste  
471993       teste  
471994       teste  
471995       teste  

[39732 rows x 18 columns]

In [ ]:
covid2 = covid.copy()
covid2

regiao estado municipio  coduf    codmun  codRegiaoSaude  \
0             Brasil    NaN       NaN     76       NaN             NaN   
1             Brasil    NaN       NaN     76       NaN             NaN   
2             Brasil    NaN       NaN     76       NaN             NaN   
3             Brasil    NaN       NaN     76       NaN             NaN   
4             Brasil    NaN       NaN     76       NaN             NaN   
...              ...    ...       ...    ...       ...             ...   
471991  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471992  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471993  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471994  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471995  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   

         nomeRegiaoSaude        data  semanaEpi  populacaoTCU2019  \
0                    NaN  2022-07-01         26       210147125.0   
1                    NaN  2022-07-02         26       210147125.0   
2                    NaN  2022-07-03         27       210147125.0   
3                    NaN  2022-07-04         27       210147125.0   
4                    NaN  2022-07-05         27       210147125.0   
...                  ...         ...        ...               ...   
471991  DISTRITO FEDERAL  2022-09-18         38         3015268.0   
471992  DISTRITO FEDERAL  2022-09-19         38         3015268.0   
471993  DISTRITO FEDERAL  2022-09-20         38         3015268.0   
471994  DISTRITO FEDERAL  2022-09-21         38         3015268.0   
471995  DISTRITO FEDERAL  2022-09-22         38         3015268.0   

        casosAcumulado  casosNovos  obitosAcumulado  obitosNovos  \
0             32434063       76045           671700          284   
1             32471847       37784           671858          158   
2             32490422       18575           671911           53   
3             32535923       45501           672033          122   
4             32610514       74591           672429          396   
...                ...         ...              ...          ...   
471991          838413           0            11825            0   
471992          838501          88            11825            0   
471993          838530          29            11828            3   
471994          838573          43            11829            1   
471995          838619          46            11829            0   

        Recuperadosnovos  emAcompanhamentoNovos  interior/metropolitana  
0             30873682.0               888681.0                     NaN  
1             30880584.0               919405.0                     NaN  
2             30906575.0               911936.0                     NaN  
3             30967114.0               896776.0                     NaN  
4             31039055.0               899030.0                     NaN  
...                  ...                    ...                     ...  
471991               NaN                    NaN                     1.0  
471992               NaN                    NaN                     1.0  
471993               NaN                    NaN                     1.0  
471994               NaN                    NaN                     1.0  
471995               NaN                    NaN                     1.0  

[471996 rows x 17 columns]

In [ ]:
covid2.loc[0:10, 'regiao'] = 'teste'
covid2.head()

regiao estado municipio  coduf  codmun  codRegiaoSaude nomeRegiaoSaude  \
0  teste    NaN       NaN     76     NaN             NaN             NaN   
1  teste    NaN       NaN     76     NaN             NaN             NaN   
2  teste    NaN       NaN     76     NaN             NaN             NaN   
3  teste    NaN       NaN     76     NaN             NaN             NaN   
4  teste    NaN       NaN     76     NaN             NaN             NaN   

         data  semanaEpi  populacaoTCU2019  casosAcumulado  casosNovos  \
0  2022-07-01         26       210147125.0        32434063       76045   
1  2022-07-02         26       210147125.0        32471847       37784   
2  2022-07-03         27       210147125.0        32490422       18575   
3  2022-07-04         27       210147125.0        32535923       45501   
4  2022-07-05         27       210147125.0        32610514       74591   

   obitosAcumulado  obitosNovos  Recuperadosnovos  emAcompanhamentoNovos  \
0           671700          284        30873682.0               888681.0   
1           671858          158        30880584.0               919405.0   
2           671911           53        30906575.0               911936.0   
3           672033          122        30967114.0               896776.0   
4           672429          396        31039055.0               899030.0   

   interior/metropolitana (0, regiao)  
0                     NaN       teste  
1                     NaN       teste  
2                     NaN       teste  
3                     NaN       teste  
4                     NaN       teste

In [ ]:
covid.iloc[0].to_frame()

0
regiao                       Brasil
estado                          NaN
municipio                       NaN
coduf                            76
codmun                          NaN
codRegiaoSaude                  NaN
nomeRegiaoSaude                 NaN
data                     2022-07-01
semanaEpi                        26
populacaoTCU2019        210147125.0
casosAcumulado             32434063
casosNovos                    76045
obitosAcumulado              671700
obitosNovos                     284
Recuperadosnovos         30873682.0
emAcompanhamentoNovos      888681.0
interior/metropolitana          NaN
(0, regiao)                   teste

In [ ]:
covid.iloc[0,0]

'Brasil'

In [ ]:
covid.iloc[0].to_frame().T

regiao estado municipio coduf codmun codRegiaoSaude nomeRegiaoSaude  \
0  Brasil    NaN       NaN    76    NaN            NaN             NaN   

         data semanaEpi populacaoTCU2019 casosAcumulado casosNovos  \
0  2022-07-01        26      210147125.0       32434063      76045   

  obitosAcumulado obitosNovos Recuperadosnovos emAcompanhamentoNovos  \
0          671700         284       30873682.0              888681.0   

  interior/metropolitana (0, regiao)  
0                    NaN       teste

In [ ]:
covid.iloc[-5:, ]

regiao estado municipio  coduf    codmun  codRegiaoSaude  \
471991  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471992  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471993  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471994  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471995  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   

         nomeRegiaoSaude        data  semanaEpi  populacaoTCU2019  \
471991  DISTRITO FEDERAL  2022-09-18         38         3015268.0   
471992  DISTRITO FEDERAL  2022-09-19         38         3015268.0   
471993  DISTRITO FEDERAL  2022-09-20         38         3015268.0   
471994  DISTRITO FEDERAL  2022-09-21         38         3015268.0   
471995  DISTRITO FEDERAL  2022-09-22         38         3015268.0   

        casosAcumulado  casosNovos  obitosAcumulado  obitosNovos  \
471991          838413           0            11825            0   
471992          838501          88            11825            0   
471993          838530          29            11828            3   
471994          838573          43            11829            1   
471995          838619          46            11829            0   

        Recuperadosnovos  emAcompanhamentoNovos  interior/metropolitana  \
471991               NaN                    NaN                     1.0   
471992               NaN                    NaN                     1.0   
471993               NaN                    NaN                     1.0   
471994               NaN                    NaN                     1.0   
471995               NaN                    NaN                     1.0   

       (0, regiao)  
471991       teste  
471992       teste  
471993       teste  
471994       teste  
471995       teste

In [ ]:
covid['estado'].value_counts()

MG    71820
SP    54348
RS    41916
BA    35196
PR    33684
SC    24948
GO    20832
PI    18984
PB    18900
MA    18396
PE    15708
CE    15624
RN    14196
PA    12180
MT    12012
TO    11844
AL     8736
RJ     7896
MS     6720
ES     6720
SE     6468
AM     5292
RO     4536
AC     1932
AP     1428
RR     1428
DF      168
Name: estado, dtype: int64

# 1 - Questão
Utilizando a função loc, filtre toda tabela, porém somente aonde o estado for o ceará (CE)

In [ ]:
ce = covid.loc[covid['estado'] == 'CE', :]
ce

regiao estado        municipio  coduf    codmun  codRegiaoSaude  \
840    Nordeste     CE              NaN     23       NaN             NaN   
841    Nordeste     CE              NaN     23       NaN             NaN   
842    Nordeste     CE              NaN     23       NaN             NaN   
843    Nordeste     CE              NaN     23       NaN             NaN   
844    Nordeste     CE              NaN     23       NaN             NaN   
...         ...    ...              ...    ...       ...             ...   
93151  Nordeste     CE  Viçosa do Ceará     23  231410.0         23013.0   
93152  Nordeste     CE  Viçosa do Ceará     23  231410.0         23013.0   
93153  Nordeste     CE  Viçosa do Ceará     23  231410.0         23013.0   
93154  Nordeste     CE  Viçosa do Ceará     23  231410.0         23013.0   
93155  Nordeste     CE  Viçosa do Ceará     23  231410.0         23013.0   

          nomeRegiaoSaude        data  semanaEpi  populacaoTCU2019  \
840                   NaN  2022-07-01         26         9132078.0   
841                   NaN  2022-07-02         26         9132078.0   
842                   NaN  2022-07-03         27         9132078.0   
843                   NaN  2022-07-04         27         9132078.0   
844                   NaN  2022-07-05         27         9132078.0   
...                   ...         ...        ...               ...   
93151  13ª REGIAO TIANGUA  2022-09-18         38           60889.0   
93152  13ª REGIAO TIANGUA  2022-09-19         38           60889.0   
93153  13ª REGIAO TIANGUA  2022-09-20         38           60889.0   
93154  13ª REGIAO TIANGUA  2022-09-21         38           60889.0   
93155  13ª REGIAO TIANGUA  2022-09-22         38           60889.0   

       casosAcumulado  casosNovos  obitosAcumulado  obitosNovos  \
840           1267955        1556            27200           14   
841           1269537        1582            27212           12   
842           1270401         864            27216            4   
843           1270625         224            27216            0   
844           1272422        1797            27227           11   
...               ...         ...              ...          ...   
93151            6426           0              163            0   
93152            6426           0              163            0   
93153            6426           0              163            0   
93154            6429           3              163            0   
93155            6429           0              163            0   

       Recuperadosnovos  emAcompanhamentoNovos  interior/metropolitana  \
840                 NaN                    NaN                     NaN   
841                 NaN                    NaN                     NaN   
842                 NaN                    NaN                     NaN   
843                 NaN                    NaN                     NaN   
844                 NaN                    NaN                     NaN   
...                 ...                    ...                     ...   
93151               NaN                    NaN                     0.0   
93152               NaN                    NaN                     0.0   
93153               NaN                    NaN                     0.0   
93154               NaN                    NaN                     0.0   
93155               NaN                    NaN                     0.0   

      (0, regiao)  
840         teste  
841         teste  
842         teste  
843         teste  
844         teste  
...           ...  
93151       teste  
93152       teste  
93153       teste  
93154       teste  
93155       teste  

[15624 rows x 18 columns]

# 2 - Questão
Agora com o estado filtrado pegue o municipio de fortaleza

In [ ]:
ce.loc[ce['municipio'] == 'Fortaleza']

regiao estado  municipio  coduf    codmun  codRegiaoSaude  \
82572  Nordeste     CE  Fortaleza     23  230440.0         23001.0   
82573  Nordeste     CE  Fortaleza     23  230440.0         23001.0   
82574  Nordeste     CE  Fortaleza     23  230440.0         23001.0   
82575  Nordeste     CE  Fortaleza     23  230440.0         23001.0   
82576  Nordeste     CE  Fortaleza     23  230440.0         23001.0   
...         ...    ...        ...    ...       ...             ...   
82651  Nordeste     CE  Fortaleza     23  230440.0         23001.0   
82652  Nordeste     CE  Fortaleza     23  230440.0         23001.0   
82653  Nordeste     CE  Fortaleza     23  230440.0         23001.0   
82654  Nordeste     CE  Fortaleza     23  230440.0         23001.0   
82655  Nordeste     CE  Fortaleza     23  230440.0         23001.0   

           nomeRegiaoSaude        data  semanaEpi  populacaoTCU2019  \
82572  1ª REGIAO FORTALEZA  2022-07-01         26         2669342.0   
82573  1ª REGIAO FORTALEZA  2022-07-02         26         2669342.0   
82574  1ª REGIAO FORTALEZA  2022-07-03         27         2669342.0   
82575  1ª REGIAO FORTALEZA  2022-07-04         27         2669342.0   
82576  1ª REGIAO FORTALEZA  2022-07-05         27         2669342.0   
...                    ...         ...        ...               ...   
82651  1ª REGIAO FORTALEZA  2022-09-18         38         2669342.0   
82652  1ª REGIAO FORTALEZA  2022-09-19         38         2669342.0   
82653  1ª REGIAO FORTALEZA  2022-09-20         38         2669342.0   
82654  1ª REGIAO FORTALEZA  2022-09-21         38         2669342.0   
82655  1ª REGIAO FORTALEZA  2022-09-22         38         2669342.0   

       casosAcumulado  casosNovos  obitosAcumulado  obitosNovos  \
82572          369868         387            11307            2   
82573          370221         353            11311            4   
82574          370565         344            11315            4   
82575          370647          82            11315            0   
82576          371138         491            11320            5   
...               ...         ...              ...          ...   
82651          391537          -9            11494            0   
82652          391552          15            11494            0   
82653          391550          -2            11497            3   
82654          391556           6            11503            6   
82655          391519         -37            11504            1   

       Recuperadosnovos  emAcompanhamentoNovos  interior/metropolitana  \
82572               NaN                    NaN                     1.0   
82573               NaN                    NaN                     1.0   
82574               NaN                    NaN                     1.0   
82575               NaN                    NaN                     1.0   
82576               NaN                    NaN                     1.0   
...                 ...                    ...                     ...   
82651               NaN                    NaN                     1.0   
82652               NaN                    NaN                     1.0   
82653               NaN                    NaN                     1.0   
82654               NaN                    NaN                     1.0   
82655               NaN                    NaN                     1.0   

      (0, regiao)  
82572       teste  
82573       teste  
82574       teste  
82575       teste  
82576       teste  
...           ...  
82651       teste  
82652       teste  
82653       teste  
82654       teste  
82655       teste  

[84 rows x 18 columns]

# 3 - Questão 
Filtre os estados que estejam com a semana epidemiológica acima de 30 __e__ que sejam do Distrito Federal(DF)

In [ ]:
df = covid.loc[(covid['semanaEpi'] > 30) & (covid['estado'] == 'DF')]
df

regiao estado municipio  coduf    codmun  codRegiaoSaude  \
2298    Centro-Oeste     DF       NaN     53       NaN             NaN   
2299    Centro-Oeste     DF       NaN     53       NaN             NaN   
2300    Centro-Oeste     DF       NaN     53       NaN             NaN   
2301    Centro-Oeste     DF       NaN     53       NaN             NaN   
2302    Centro-Oeste     DF       NaN     53       NaN             NaN   
...              ...    ...       ...    ...       ...             ...   
471991  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471992  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471993  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471994  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   
471995  Centro-Oeste     DF  Brasília     53  530010.0         53001.0   

         nomeRegiaoSaude        data  semanaEpi  populacaoTCU2019  \
2298                 NaN  2022-07-31         31         3015268.0   
2299                 NaN  2022-08-01         31         3015268.0   
2300                 NaN  2022-08-02         31         3015268.0   
2301                 NaN  2022-08-03         31         3015268.0   
2302                 NaN  2022-08-04         31         3015268.0   
...                  ...         ...        ...               ...   
471991  DISTRITO FEDERAL  2022-09-18         38         3015268.0   
471992  DISTRITO FEDERAL  2022-09-19         38         3015268.0   
471993  DISTRITO FEDERAL  2022-09-20         38         3015268.0   
471994  DISTRITO FEDERAL  2022-09-21         38         3015268.0   
471995  DISTRITO FEDERAL  2022-09-22         38         3015268.0   

        casosAcumulado  casosNovos  obitosAcumulado  obitosNovos  \
2298            830565           0            11819            0   
2299            831184         619            11820            1   
2300            831776         592            11821            1   
2301            832270         494            11821            0   
2302            832723         453            11822            1   
...                ...         ...              ...          ...   
471991          838413           0            11825            0   
471992          838501          88            11825            0   
471993          838530          29            11828            3   
471994          838573          43            11829            1   
471995          838619          46            11829            0   

        Recuperadosnovos  emAcompanhamentoNovos  interior/metropolitana  \
2298                 NaN                    NaN                     NaN   
2299                 NaN                    NaN                     NaN   
2300                 NaN                    NaN                     NaN   
2301                 NaN                    NaN                     NaN   
2302                 NaN                    NaN                     NaN   
...                  ...                    ...                     ...   
471991               NaN                    NaN                     1.0   
471992               NaN                    NaN                     1.0   
471993               NaN                    NaN                     1.0   
471994               NaN                    NaN                     1.0   
471995               NaN                    NaN                     1.0   

       (0, regiao)  
2298         teste  
2299         teste  
2300         teste  
2301         teste  
2302         teste  
...            ...  
471991       teste  
471992       teste  
471993       teste  
471994       teste  
471995       teste  

[108 rows x 18 columns]

# 4 - Questão
Some agora a coluna de "obitosNovos" do distrito federal

In [ ]:
df['obitosNovos'].sum()

20

In [ ]:
covid.describe()

coduf         codmun  codRegiaoSaude      semanaEpi  \
count  471996.000000  469644.000000   467880.000000  471996.000000   
mean       32.360384  325258.014130    32403.123698      32.214286   
std         9.873307   98535.129104     9836.351818       3.481490   
min        11.000000  110000.000000    11001.000000      26.000000   
25%        25.000000  251200.000000    25010.000000      29.000000   
50%        31.000000  314610.000000    31059.000000      32.000000   
75%        41.000000  411915.000000    41015.000000      35.000000   
max        76.000000  530010.000000    53001.000000      38.000000   

       populacaoTCU2019  casosAcumulado     casosNovos  obitosAcumulado  \
count      4.702320e+05    4.719960e+05  471996.000000    471996.000000   
mean       1.126190e+05    1.810664e+04      14.355865       363.115107   
std        2.935317e+06    4.701238e+05     511.199512      9572.018291   
min        7.810000e+02    0.000000e+00  -30211.000000         0.000000   
25%        5.474000e+03    7.660000e+02       0.000000        11.000000   
50%        1.168900e+04    1.596000e+03       0.000000        24.000000   
75%        2.576200e+04    3.909000e+03       1.000000        62.000000   
max        2.101471e+08    3.461666e+07  107959.000000    685677.000000   

         obitosNovos  Recuperadosnovos  emAcompanhamentoNovos  \
count  471996.000000      8.300000e+01           8.300000e+01   
mean        0.090643      3.264563e+07           5.809355e+05   
std         4.398072      9.346696e+05           3.239098e+05   
min      -762.000000      3.087368e+07           1.422540e+05   
25%         0.000000      3.186199e+07           2.698635e+05   
50%         0.000000      3.292776e+07           5.177680e+05   
75%         0.000000      3.348802e+07           9.099985e+05   
max       762.000000      3.378872e+07           1.084419e+06   

       interior/metropolitana  
count           467880.000000  
mean                 0.069300  
std                  0.253964  
min                  0.000000  
25%                  0.000000  
50%                  0.000000  
75%                  0.000000  
max                  1.000000

In [ ]:
data = {'turmas':['1A','1A','2A','2A','3A','3A','3A'],
        'alunos': ['Carlos','Grazi','Nator', 'Juci', 'Miguel', 'Aline', 'Emanuel'],
        'notas':[7,8,10,6,5,4.5,10]
        }
        
escola = pd.DataFrame(data)
escola

turmas   alunos  notas
0     1A   Carlos    7.0
1     1A    Grazi    8.0
2     2A    Nator   10.0
3     2A     Juci    6.0
4     3A   Miguel    5.0
5     3A    Aline    4.5
6     3A  Emanuel   10.0

In [ ]:
escola.mean()

<ipython-input-106-653bde871b38>:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  escola.mean()


notas    7.214286
dtype: float64

In [ ]:
escola.groupby('turmas').count()

alunos  notas
turmas               
1A           2      2
2A           2      2
3A           3      3

In [ ]:
escola.groupby("turmas").mean()

notas
turmas       
1A        7.5
2A        8.0
3A        6.5

In [ ]:
paises = pd.DataFrame({
    "country" : ["United States", "Canada", "United Kingdom", "France", "Germany", "China", "Japan", "South Korea"],
    "continent" : ["North America", "North America", "Europe", "Europe", "Europe", "Asia", "Asia", "Asia"],
    "population" : [332722557, 38711108, 67081234, 67853000, 83222442, 1412600000, 125502000, 51745000],
    "area" : [3796742, 3855100, 93628, 247368, 137882, 3705407, 145937, 38690],
    "population percentage of world": [4.18, 0.487, 0.843, 0.853, 1.05, 17.8, 1.58, 0.650]
})

paises

country      continent  population     area  \
0   United States  North America   332722557  3796742   
1          Canada  North America    38711108  3855100   
2  United Kingdom         Europe    67081234    93628   
3          France         Europe    67853000   247368   
4         Germany         Europe    83222442   137882   
5           China           Asia  1412600000  3705407   
6           Japan           Asia   125502000   145937   
7     South Korea           Asia    51745000    38690   

   population percentage of world  
0                           4.180  
1                           0.487  
2                           0.843  
3                           0.853  
4                           1.050  
5                          17.800  
6                           1.580  
7                           0.650

In [ ]:
paises.groupby('continent').sum()

population     area  population percentage of world
continent                                                         
Asia           1589847000  3890034                          20.030
Europe          218156676   478878                           2.746
North America   371433665  7651842                           4.667

In [ ]:
paises.groupby('continent')[['population']].sum()

population
continent                
Asia           1589847000
Europe          218156676
North America   371433665

In [ ]:
paises.groupby('continent')[['population','area']].sum().sort_values(by='population', ascending=False)

population     area
continent                         
Asia           1589847000  3890034
North America   371433665  7651842
Europe          218156676   478878

In [ ]:
loja = pd.read_csv('https://raw.githubusercontent.com/rafaelpuyau/infinity_school/main/ds/datasets/atividade_casa.csv')
loja

Produto      Cor       Valor  Quantidade       Sexo  Idade  \
0         Calça     Roxa   66.947509           4  Masculino     39   
1      Camiseta  Amarela   99.692292           1  Masculino     30   
2         Calça     Azul   92.502413           2   Feminino     25   
3         Short     Azul  162.693470           1   Feminino     45   
4      Camiseta  Laranja  148.191049           3   Feminino     29   
...         ...      ...         ...         ...        ...    ...   
99995     Calça  Laranja  101.158897           4  Masculino     46   
99996      Saia    Preta  100.815311           3  Masculino     28   
99997     Calça     Azul   63.924217           4   Feminino     48   
99998     Calça     Azul  149.062392           3   Feminino     19   
99999      Saia    Verde  114.089710           3  Masculino     25   

      Data da Venda         Canal  
0        2022-02-25  Loja Virtual  
1        2022-01-19   Loja Física  
2        2022-03-11   Loja Física  
3        2022-05-01   Loja Física  
4        2022-03-13   Loja Física  
...             ...           ...  
99995    2022-01-24   Loja Física  
99996    2022-03-08   Loja Física  
99997    2022-06-08  Loja Virtual  
99998    2022-01-05   Loja Física  
99999    2022-04-19  Loja Virtual  

[100000 rows x 8 columns]

In [ ]:
resultado = loja.groupby(['Produto', 'Cor']).agg( {'Valor': 'max', 'Idade': 
                                       ['sum','min','mean']} )
resultado

Valor  Idade               
                          max    sum min       mean
Produto  Cor                                       
Calça    Amarela   179.983554  83153  18  33.570044
         Azul      179.879813  85554  18  33.186191
         Cinza     179.999629  83744  18  33.271355
         Laranja   179.934841  88638  18  33.715481
         Preta     179.844980  83723  18  33.610197
         Roxa      179.996657  82503  18  33.280758
         Verde     179.960123  83441  18  33.363055
         Vermelha  179.945384  85990  18  33.602970
Camisa   Amarela   179.917138  82451  18  33.327001
         Azul      179.940290  83650  18  33.486789
         Cinza     179.953572  79980  18  33.408521
         Laranja   179.979988  85147  18  33.628357
         Preta     179.985823  83218  18  33.528606
         Roxa      179.962381  83181  18  33.594911
         Verde     179.901529  84768  18  33.086651
         Vermelha  179.952399  83133  18  33.173583
Camiseta Amarela   179.946660  82405  18  33.648428
         Azul      179.938695  85024  18  33.699564
         Cinza     179.904847  82680  18  33.325272
         Laranja   179.927107  83994  18  33.517159
         Preta     179.950929  80568  18  33.710460
         Roxa      179.962973  79865  18  33.360485
         Verde     179.944304  86174  18  33.609204
         Vermelha  179.960038  83699  18  33.506405
Saia     Amarela   179.824738  85990  18  33.407148
         Azul      179.999542  84095  18  33.490641
         Cinza     179.908102  82576  18  33.458671
         Laranja   179.889852  84717  18  33.471750
         Preta     179.858712  83466  18  33.480144
         Roxa      179.991598  80735  18  33.210613
         Verde     179.986046  82025  18  33.452284
         Vermelha  179.905564  83806  18  33.256349
Short    Amarela   179.836905  82866  18  33.454178
         Azul      179.987474  84967  18  33.464750
         Cinza     179.976456  83369  18  33.214741
         Laranja   179.913663  83904  18  33.534772
         Preta     179.953965  82936  18  33.361223
         Roxa      179.987294  86210  18  33.427685
         Verde     179.992850  83859  18  33.237812
         Vermelha  179.896695  81893  18  33.507774

# 1 - Questão

Agrupe os produtos e veja qual a média de valor cada um deles.

produtos nessa tabela:

*   Calça
*   Camisa	
*   Camiseta
*   Saia	
*   Short


In [ ]:
loja.groupby('Produto')[['Valor']].mean()

Valor
Produto             
Calça     114.944747
Camisa    115.817377
Camiseta  115.103593
Saia      114.897358
Short     115.201277

# 2 - Questão
Agrupe as roupas por cor e veja o somatorio da quantidade de roupas daquela cor respectiva. Em suma faça a quantidade de cada cor.

In [ ]:
loja.groupby('Cor')[['Quantidade']].sum()

Quantidade
Cor                 
Amarela        31059
Azul           31574
Cinza          30924
Laranja        31891
Preta          30904
Roxa           30970
Verde          31672
Vermelha       31433

# 3 - Questão

Veja a média de idade de homens e mulheres que compraram na loja


In [ ]:
loja.groupby('Sexo')[['Idade']].mean()

Idade
Sexo               
Feminino   33.44073
Masculino  33.44121

# 5 - Questão

Filtre os __estados__ de São Paulo e o Estado de Minas Gerais e depois veja quantas vezes os estados apareceram

Resposta:
MG -> 71820
<br>
SP -> 54348

In [ ]:
covid.loc[(covid['estado'] == 'MG') | (covid['estado'] == 'SP')]['estado'].value_counts()

MG    71820
SP    54348
Name: estado, dtype: int64